lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -321,21 +326,22 @@ impl PartialEq for WeakLayer {
 
 /// Internal representation of [`Layer`].
 ///
-/// Please note that the [`parents`] field contains reference to a very small part of parent layer,
+/// Please note that the [`parent`] field contains reference to a very small part of parent layer,
 /// namely to its [`Sublayers`] struct. Only this part is needed to properly update all the models.
-#[derive(Clone)]
 #[allow(missing_docs)]
 pub struct LayerModel {
     pub name: String,
-    pub camera: RefCell<Camera2d>,
+    camera: RefCell<Camera2d>,
+    inherit_camera: bool,
```

Comment:
No explanation of how these two fields work. Camera looks like this layer has alwyas camera assigned, so in such a case, how inherit_camera works? Is `camera` field then ignored?

Reply:
`camera` is always used during rendering - we don't want to needlessly traverse the tree for each rendered layer. The propagation happens at camera set time, which is a way less frequent operation than rendering. The flag determines if that propagation should happen.